scheds/rust/scx_bpfland/src/bpf/main.bpf.c

```diff
@@ -965,6 +1010,22 @@ void BPF_STRUCT_OPS(bpfland_stopping, struct task_struct *p, bool runnable)
 	if (tctx->is_interactive)
 		__sync_fetch_and_sub(&nr_interactive, 1);
 
+	/*
+	 * Update task's average runtime.
+	 */
+	slice = now - tctx->last_run_at;
+	tctx->sum_runtime += slice;
+	tctx->avg_runtime = calc_avg(tctx->avg_runtime, tctx->sum_runtime);
+
+	/*
+	 * Update task vruntime charging the weighted used time slice.
+	 */
+	p->scx.dsq_vtime += scale_inverse_fair(p, tctx, slice);
```

Comment:
@hodgesds if I'm not mistaked ops.stopping() is also called when a task calls `sched_yield()`, and the fact that yield is implemented by setting its time slice to zero it should follow the same pattern as a task using all its assigned time slice.

And since we measure the used time doing a diff between the timestamp at `ops.stopping()` - the timestamp at `ops.running()`, we should get an accurate used time independently on how much `p->scx.slice` has been used.

Does it make sense? Am I missing something?

Reply:
That makes sense, thanks for the explaination!